app/billing/api.go

```diff
@@ -141,3 +153,59 @@ func (h *handlers) handleCharge(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+func (h *handlers) handleGetStats(w http.ResponseWriter, _ *http.Request) {
+	resp, err := h.temporal.DescribeTaskQueueEnhanced(context.Background(), client.DescribeTaskQueueEnhancedOptions{
+		TaskQueue:     TaskQueue,
+		ReportStats:   true,
+		ReportPollers: true,
+	})
+	if err != nil {
+		h.logger.Error("Failed to get task queue stats", "error", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	recentPollerWindow := time.Now().Add(-1 * time.Minute)
+
+	var backlog int64
+	var workerCount int
+	for _, versionInfo := range resp.VersionsInfo {
+		for _, typeInfo := range versionInfo.TypesInfo {
+			for _, pollerInfo := range typeInfo.Pollers {
+				if pollerInfo.LastAccessTime.After(recentPollerWindow) {
+					workerCount++
+				}
+			}
+			if typeInfo.Stats != nil {
+				backlog += typeInfo.Stats.ApproximateBacklogCount
+			}
+		}
+	}
+
+	closedSince := time.Now().Add(-statsInterval * time.Second).Format(time.RFC3339)
+	countResp, err := h.temporal.CountWorkflow(context.Background(), &workflowservice.CountWorkflowExecutionsRequest{
+		Query: fmt.Sprintf("WorkflowType='Charge' AND ExecutionStatus='Completed' AND CloseTime > %q", closedSince),
+	})
```

Comment:
I think this is too eventually consistent to rely upon at 30s intervals. Cloud has some real visibility latency for workflow execution status completed update frequency we've observed. I don't think Temporal today offers a way to get workflow completion rate for recently completed workflows. If this information is needed real time it should be out of band.